power-adapters-data/src/main/java/com/nextfaze/poweradapters/data/CursorData.java

```diff
@@ -0,0 +1,91 @@
+package com.nextfaze.poweradapters.data;
+
+import android.database.Cursor;
+import android.support.annotation.NonNull;
+
+import java.io.Closeable;
+
+/**
+ * Simple {@link Data} implementation backed by a {@link Cursor}. The cursor can be changed or replaced at any time.
+ * The cursor will never be closed.
+ *
+ * @param <T> The type of element this data contains.
+ */
+public abstract class CursorData<T> extends Data<T> implements Closeable {
+
+    @NonNull
+    private Cursor mData;
+
+    public CursorData() {
+    }
+
+    /**
+     * @return A new element without any data from the cursor.
+     */
+    public abstract T newElement();
```

Comment:
Rather than `newElement` and `populateElement`, which assume your models are mutable, I'm more inclined to have a function that simply maps a `Cursor` row to `T`. That way you can use immutable models (which I recommend), and it only requires a single method.